src/coreclr/vm/siginfo.cpp

```diff
@@ -387,6 +387,50 @@ void SigPointer::ConvertToInternalSignature(Module* pSigModule, SigTypeContext *
     }
 }
 
+void SigPointer::CopyModOptsReqs(SigBuilder * pSigBuilder)
+{
+    CONTRACTL
+    {
+        INSTANCE_CHECK;
+        STANDARD_VM_CHECK;
+    }
+    CONTRACTL_END
+
+    CorElementType typ;
+    IfFailThrow(PeekElemType(&typ));
+    while (typ == ELEMENT_TYPE_CMOD_REQD || typ == ELEMENT_TYPE_CMOD_OPT)
+    {
+        // Skip the custom modifier
+        IfFailThrow(GetByte(NULL));
+
+        // Get the encoded token.
+        uint32_t token;
+        IfFailThrow(GetToken(&token));
+
+        // Append the custom modifier and encoded token to the signature.
+        pSigBuilder->AppendElementType(typ);
+        pSigBuilder->AppendToken(token);
+
+        typ = ELEMENT_TYPE_END;
+        IfFailThrow(PeekElemType(&typ));
+    }
+}
+
+void SigPointer::CopyExactlyOne(SigBuilder* pSigBuilder)
+{
+    CONTRACTL
+    {
+        INSTANCE_CHECK;
+        STANDARD_VM_CHECK;
+    }
+    CONTRACTL_END
+
+    intptr_t beginExactlyOne = (intptr_t)m_ptr;
+    IfFailThrow(SkipExactlyOne());
+    intptr_t endExactlyOne = (intptr_t)m_ptr;
+    pSigBuilder->AppendBlob((const PVOID)(beginExactlyOne), endExactlyOne - beginExactlyOne);
```

Comment:
These two new Copy... methods move the pointer to the end of the thing being copies.

The existing CopySignature method keep the pointer intact.

Could you please fix the existing CopySignature method to move the pointer to the end of thing being copied as well for consistency?